test/e2e/daemonset/daemonset.go

```diff
@@ -0,0 +1,43 @@
+package daemonset
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type DaemonSetCommands struct{ daemonSet appsv1.DaemonSet }
+
+func (dc DaemonSetCommands) ReadyConditionEventually() AsyncAssertion {
```

Comment:
We have suffer a lot of not doing proper asserting at Ginkgo and hidden matching details, so is very difficult to debug errors, so let's prepare a custom matcher so we can log all we need on failure and plug it nicely with expectations.

So at matchers.go under this `daemonset` package we can add the following matcher to check daemonsetReadinness:

```golang
package daemonset                                                               
                                                                                
import (                                                                        
    "fmt"                                                                       
                                                                                
    "github.com/onsi/gomega/format"                                             
    "github.com/onsi/gomega/types"                                              
                                                                                
    appsv1 "k8s.io/api/apps/v1"                                                 
)                                                                               
                                                                                
func BeReady() types.GomegaMatcher {                                            
    return &isReady{}                                                           
}                                                                               
                                                                                
type isReady struct {                                                           
    obtainedDaemonSet *appsv1.DaemonSet                                         
}                                                                               
                                                                                
func (matcher *isReady) Match(obtained interface{}) (success bool, err error) { 
    obtainedDaemonSet, ok := obtained.(*appsv1.DaemonSet)                       
    if !ok {                                                                    
        return false, fmt.Errorf("daemonset.IsReady matcher expects an appvs1.DaemonSet")
    }                                                                           
    matcher.obtainedDaemonSet = obtainedDaemonSet                               
    return matcher.expectedNumberOfPods() == matcher.availableNumberOfPods(), nil
}                                                                               
                                                                                
func (matcher *isReady) FailureMessage(actual interface{}) (message string) {   
    return matcher.message("to equal")                                          
}                                                                               
                                                                                
func (matcher *isReady) NegatedFailureMessage(actual interface{}) (message string) {
    return matcher.message("to not equal")                                      
}                                                                               
                                                                                
func (matcher *isReady) message(message string) string {                        
    return format.Message(matcher.expectedNumberOfPods(), fmt.Sprintf("daemonset.Status.DesiredNumberScheduled %s daemonset.Status.NumberAvailable", message), matcher.availableNumberOfPods())
}                                                                               
                                                                                
func (matcher *isReady) expectedNumberOfPods() int32 {                          
    return matcher.obtainedDaemonSet.Status.DesiredNumberScheduled              
}                                                                               
                                                                                
func (matcher *isReady) availableNumberOfPods() int32 {                         
    return matcher.obtainedDaemonSet.Status.NumberAvailable                     
}                                                                               
```